### DISCUSSION TITLE
Can I use CREATE DATABASE ... TEMPLATE for databases with pgroonga?

### DISCUSSION THREAD

**Discussion by MMeent** [pgroonga/pgroonga#439] [Q&A]
PostgreSQL `CREATE DATABASE` has a `template` argument which defaults to `template1`, where data from one database is copied from one database to the other.

As PGRoonga doesn't use Postgres' buffer manager and page format, but instead uses its own storage system, how does this interact with `CREATE DATABASE ... TEMPLATE my_existing_database_with_pgroonga_indexes`?  
Can I expect it to work out-of-the-box, or are there special considerations I should keep in mind?

**Comment by askdkc** [comment]
I'm not an expert but if you want to create a database from existing DB as template, you can do

```bash
createdb newdb -T YOUR_EXISTING_DB_WITH_PGROONGA_INDEX
```

But the problem with this approach is that newdb doesn't have PGRoonga index as expected so you need to 

```psql
REINDEX DATABASE newdb;
```

And you're ready.
